tests/ui/matches.stderr

```diff
@@ -426,3 +426,25 @@ note: consider refactoring into `Ok(3) | Ok(_)`
     |                  ^^^^^^^^^^^^^^
     = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
 
+error: use as_ref() instead
+   --> $DIR/matches.rs:320:20
+    |
+320 |       let borrowed = match owned {
+    |  ____________________^
+321 | |         None => None,
+322 | |         Some(ref v) => Some(v),
+323 | |     };
+    | |_____^ help: try this: `owned.as_ref()`
+    |
+    = note: `-D match-as-ref` implied by `-D warnings`
+
+error: use as_ref() instead
+   --> $DIR/matches.rs:326:28
+    |
+326 |       let mut mut_borrowed = match mut_owned {
+    |  ____________________________^
+327 | |         None => None,
+328 | |         Some(ref mut v) => Some(v),
+329 | |     };
+    | |_____^ help: try this: `mut_owned.as_ref()`
```

Comment:
This should be suggesting `mut_owned.as_mut()`

Reply:
Good catch, I didn't think of `Option<&mut T>` values